### PR TITLE
Assert when ProjectCommand's aren't marked with [ProjectCommandAttribute]

### PR DESCRIPTION
**Pull request by davkean**
tag @dotnet/roslyn-analysis

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Designers/Input/AbstractProjectCommand.cs [hunk 1]
```diff
@@ -52,8 +52,9 @@ public Task<bool> TryHandleCommandAsync(IImmutableSet<IProjectTree> nodes, long
         private static long[] GetCommandIds(AbstractProjectCommand command)
         {
             ProjectCommandAttribute attribute = (ProjectCommandAttribute)Attribute.GetCustomAttribute(command.GetType(), typeof(ProjectCommandAttribute));
-            if (attribute == null)
-                return Array.Empty<long>();
+
+            // All ProjectCommand's should be marked with [ProjectCommandAttribute]
+            Assumes.NotNull(attribute);
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Designers/Input/AbstractProjectCommand.cs]: Is there a overload of Assumes.NotNull that takes a message as well?

**Comment by srivatsn** [comment]
👍

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Designers/Input/AbstractProjectCommand.cs (hunk 1 above)
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Designers/Input/AbstractProjectCommand.cs]: Nope, I think we'll want to eventually hook up asserts like this to Watson - how does Roslyn handle this?

Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Designers/Input/AbstractProjectCommand.cs]: Roslyn initiates a non-fatal Watson that doesn't crash the process but sends the dumps.

**Pull request merged**